### ISSUE TITLE
Confusing error message on invalid input

### ISSUE THREAD

**Issue by KarolS** [jonschlinkert/is-even#6]
#### What I wrote:

    isEven("two")

#### What I expected:

    is-even expects a number.

(or something like that)

#### What I got:

    is-odd expects a number.

**Comment by LoupVaillant** [comment]
Yep, is-even should most probably have the same structure of is-odd, like this:

    module.exports = function isEven(i) {
      if (!isNumber(i)) {
        throw new TypeError('is-even expects a number.');
      }
      if (Number(i) !== Math.floor(i)) {
        throw new RangeError('is-even expects an integer.');
      }
      return !(~~i & 1);
    };

To avoid duplication, you may want to create an is-integer package:

    module.exports = function isEven(i) {
      if (!isInteger(i)) {
        throw new TypeError('is-even expects an integer.');
      }
      return !(~~i & 1);
    };

Or even have an assert that throws the exception you want directly:

    module.exports = function isEven(i) {
      assertInteger(i); // may throw a type error or a range error, depending on the exact error
      return !(~~i & 1);
    };

I'm not sold on the relevance of range error though. the odd and even predicate only make sense on integers. Giving it anything else is a type error in my book. You could simplify the above like this:

    module.exports = function isEven(i) {
      if (!isNumber(i) || Number(i) !== Math.floor(i)) {
        throw new TypeError('is-even expects an integer.');
      }
      return !(~~i & 1);
    };

If you're adamant about not duplicating `!(~~i & 1)`, then you can catch lower-level exceptions:

    module.exports = function isEven(i) {
      if (!isInteger(i)) {
        throw new RangeError('is-even expects an integer.');
      }
      try {
        return !(~~i & 1);
      } catch(err) {
        throw new TypeError('is-even expects an integer.');
    };

Maybe try to distinguish between type errors and range errors, but as I said, range error is not very relevant in my opinion. Oh, and maybe re-throw any other exception directly, though surely Javascript isn't insane enough to throw anything else there?

My, the design space is much larger than I had anticipated. Reminds me why I don't like dynamic typing—too many ways to fail.

**Comment by jonschlinkert** [comment]
> My, the design space is much larger than I had anticipated

Lol, indeed! 

> If you're adamant about not duplicating !(~~i & 1), then you can catch lower-level exceptions:

Not at all, I think we should probably re-write that code here and get rid of the dep. Fwiw, [this is what I had before](https://github.com/jonschlinkert/is-even/pull/4/files), and forgot about the error messages when I merged that PR, but the error messages and type checking are the reason for the lib. Also, I think we can just do `!(i & 1)`.

**Comment by antipodite** [comment]
lmao

**Comment by jonschlinkert** [comment]
@antipodite so, you just joined GitHub, you have no code, and your first public contribution is a troll comment. Good for you, you're getting off to a great start in the community.

**Comment by antipodite** [comment]
lmao

**Comment by jonschlinkert** [comment]
@antipodite so, you just joined GitHub, you have no code, and your first public contribution is a troll comment. Good for you, you're getting off to a great start in the community.
